File: 프로그래머스/lv2/42747. H－Index/H－Index.js
function solution(citations) {
    const count = Math.max(...citations);
    const target = citations.length;
    let answer = 0;
    
    for (let i = 0; i <= count; i++) {
        const moreThan = citations.filter((item) => item >= i).length;
        
        // h번 이상 인용된 논문이 h편 이상인가?
        if (moreThan < i) continue;
        
        const lessThan = citations.filter((item) => item < i);
        
        if (moreThan + lessThan.length !== target) continue;
        
        answer = i;
        

    }

    
    return answer;
}

File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    
    for i in range(len(citations) + 1):
        max_h_index = []
        min_h_index = []
        
        for item in citations:
            if item >= i:
                max_h_index.append(item)
            else:
                min_h_index.append(item)
        
        if len(max_h_index) >= i and len(min_h_index) <= i:
            answer = i
            
    return answer